src/ShelfCtrl/ShelfCtrl.tsx

```diff
@@ -7,15 +7,15 @@ interface Props {
 }
 
 function ShelfCtrl({ setSort }: Props) {
-  const [newShelf, setNewShelf] = useState('');
+  const [newShelf, setNewShelf] = useState<string>('');
 
-  const handleSort = (e: any,) => {
+  const handleSort = (e: React.ChangeEvent<HTMLSelectElement>): void => {
```

Comment:
Regarding the question in the PR about needing to explicitly add types in spots like this. The answer is: it depends. 

It depends on whether or not TypeScript has already inferred the type successfully or not. For example, If I remove the types on line 12 and line 18, my linter gets mad on line 12. It says, "Parameter 'e' implicitly has an 'any' type.ts". This means that your gut was right to explicitly type lines 12 and 18, since TS doesn't know what's going on - TS has not inferred the type.

However, the same isn't true of line 18. If I remove the explicit type on line 29, my linter doesn't change - there's no error. If I hover over the 'e' in the callback function, my intellisense in VSCode tells me that it knows what type it is and this means that TS has inferred the type already. In this case, we shouldn't add the type ourselves as we could unknowingly introduce an incorrect type. 

Basically, we should only fix what's broken. 

There are teams out there that insist on manually typing everything, but I've only seen them cause more issues with this. It's also possible that TS is wrong - but that's so rare and if it happens, we should first question if our preceding code is actually accurate. 

Let's chat about some other spots in the code that relate to this concept for some practice: BookProfile, Search, SearchCtrl, etc.